app/models/user.rb

```diff
@@ -1,5 +1,6 @@
 class User < ApplicationRecord
   validates :name, presence: true, uniqueness: true
+  validates :email, uniqueness: true, format: {with: /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i}
```

Comment:
what would happen if i pass in an empty email. will this validation fail? should this validation fail?

Reply:
Yes it will fail with an error message "Email is invalid" , but if we want to give an error for nil email then we can use `presence: true` before format